packages/build-tools/package.json

```diff
@@ -27,6 +27,7 @@
     "@expo/downloader": "1.0.117",
     "@expo/eas-build-job": "1.0.117",
     "@expo/env": "^0.3.0",
+    "@expo/fingerprint": "^0.9.0",
```

Comment:
I've been hesitant to hardcode this package dependency since we could see version incompatibility issues if the version used by the expo-updates in the project differs from this and changes have been made to one or the other. Put another way, since the fingerprint depends on the version of the `@expo/fingerprint` package, we generally need to ensure that any fingerprint calculations we do from the build service are done via RPC calls into the app's expo-update CLI. 

Since we're just using the `diffFingerprints` it probably doesn't matter quite as much (that function shouldn't ever need change), but the risk is that somewhere down the line we accidentally start using other functions from the package here in the build service and forget that it needs to be versioned through the expo-updates CLI.

So, my suggestion is that we don't add this dependency and instead just copy the `diffFingerprints` function into `build-tools`.

Reply:
Good idea! I wasn't a fan of adding fingerprint as a dependency either, but couldn't think of an alternative. I was intended to use the cli initially, but all the [cli options](https://github.com/expo/expo/tree/main/packages/%40expo/fingerprint#cli-usage) rely on passing in the _project path_ first so it doesn't support diffing two fingerprints. Copying the function over seems like a good compromise 👍

Reply:
Copied Kudo's `diffFingerprints` function https://github.com/expo/expo/pull/29709